### PR TITLE
Make Language a little bit more Immutable

### PR DESCRIPTION
**Pull request by TheBestPessimist**
https://github.com/metasfresh/me03/issues/1692

### CODE REVIEW

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/Env.java [hunk 1]
```diff
@@ -1585,16 +1586,17 @@ public static void verifyLanguage(final Language language)
...
 			}
 		}
 
-		// No Language - set to Base Language
-		final String baseAD_Language = Language.getBaseAD_Language();
-		s_log.warn("Not System/Base Language={} - Set to Base Language: {}", language, baseAD_Language);
-		language.setAD_Language(baseAD_Language);
-	}   // verifyLanguage
+		// If the desired language (eg. en_US requested by the browser) is unavailable, we shall return the system base language.
+		return Language.getBaseLanguage();
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/Env.java]: here why not doing the same as above?
i.e.
language.toBuilder().AD_Language(baseAD_Language).build()

File: de.metas.adempiere.adempiere/base/src/main/java/de/metas/i18n/Language.java [hunk 2]
```diff
@@ -529,6 +534,7 @@ public static Language getLoginLanguage()
 	 * @param javaDatePattern Java date pattern as not all locales are defined - if null, derived from Locale
 	 * @param mediaSize default media size
 	 */
+	@Builder(toBuilder = true)
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/base/src/main/java/de/metas/i18n/Language.java]: also make `m_AD_Language` final

Comment (TheBestPessimist) [reply de.metas.adempiere.adempiere/base/src/main/java/de/metas/i18n/Language.java]: done

File: de.metas.adempiere.adempiere/base/src/main/java/de/metas/i18n/Language.java [hunk 3]
```diff
@@ -432,7 +433,11 @@ public static Language getLanguage(@Nullable final Locale locale)
 		return getBaseLanguage();
 	}   // getLocale
 
-	public static final Language findLanguageByLocale(final Locale locale)
+	public static void addNewLanguage(Language language) {
+		s_languages.add(language);
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/base/src/main/java/de/metas/i18n/Language.java]: why not calling addIfAbsent ?

Comment (TheBestPessimist) [reply de.metas.adempiere.adempiere/base/src/main/java/de/metas/i18n/Language.java]: done

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/Env.java (hunk 1 above)
Comment (TheBestPessimist) [reply de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/Env.java]: done

**Pull request merged**